tgui/packages/tgui/interfaces/IntegratedCircuit/index.jsx

```diff
@@ -496,6 +500,21 @@ export class IntegratedCircuit extends Component {
                   }
                 />
               </Stack.Item>
+              <Stack.Item basis="24px">
+                <Button
+                  position="absolute"
+                  top={0}
+                  color="transparent"
+                  tooltip="Grid Aligning"
+                  icon="th-large"
+                  selected={isGridMode}
+                  onClick={() =>
+                    this.setState((state) => ({
+                      isGridMode: !state.isGridMode,
```

Comment:
You'll likely want to make this data sent by the server since otherwise people will have to continually turn this on any time they re-open the UI.

Reply:
check the `ui_data` proc on the DM side.
Add stuff to that list to send it to the UI. Whatever you set there will be sent whenever the UI receives any sort of update.

You can then access this data by doing `const { data } = useBackend();` and grabbing the variable from the `data` variable.
For you, this is already done on line 394, so you can just extract your variable from there and do stuff.

To update the variable, you'll need to call the `act` function to send data to the DM side.
i.e.
```js
onClick={() => act("set_grid_mode", { isGridMode = !isGridMode })}
```
All actions are sent to the `ui_act` proc, where the `action` param is the 1st arg of `act` and the 2nd argument is the key->value datatype in the 2nd arg of `act`

All aforementioned DM procs happen on the `/obj/item/integrated_circuit` type so just search for them in the integrated circuit file.

Reply:
Take a look at the `set_display_name` action if you want a simple example.